lib/index.coffee

```diff
@@ -77,32 +77,41 @@ class EPub
     @options.id = @id
     @options.images = []
     @options.content = _.map @options.content, (content, index)->
-      titleSlug = uslug removeDiacritics content.title || "no title"
-      content.filePath = path.resolve self.uuid, "./OEBPS/#{index}_#{titleSlug}.xhtml"
-      content.href = "#{index}_#{titleSlug}.xhtml"
+
+      if !content.filename
+        titleSlug = uslug removeDiacritics content.title || "no title"
+        content.href = "#{index}_#{titleSlug}.xhtml"
+        content.filePath = path.resolve self.uuid, "./OEBPS/#{index}_#{titleSlug}.xhtml"
+      else
+        content.href = if content.filename.match(/\.xhtml$/) then content.filename else "#{content.filename}.xhtml"
```

Comment:
Probably here we can bundle this into a single if and remove the console.log, like so:
```
content.href = if content.filename.match(/\.xhtml$/) then content.filename else "#{content.filename}.xhtml"
content.filePath = path.resolve self.uuid, "./OEBPS/#{content.href}"
```